### PR TITLE
Pipeline launch label support

### LINKED ISSUE #312: Add pipeline label support for 'tw launch'
Currently, we cannot add labels to pipelines using `tw launch` as you would when launching a pipeline via the UI. 

```
$ tw launch -h

Usage: tw launch [OPTIONS] PIPELINE_OR_URL

Launch a Nextflow pipeline execution.

Parameters:
*     PIPELINE_OR_URL                      Workspace pipeline name or full pipeline URL.

Options:
  -w, --workspace=<workspace>              Workspace numeric identifier (TOWER_WORKSPACE_ID as default) or workspace reference as OrganizationName/WorkspaceName
      --params-file=<paramsFile>           Pipeline parameters in either JSON or YML format.
  -c, --compute-env=<computeEnv>           Compute environment name [default: primary compute environment].
  -n, --name=<name>                        Custom workflow run name
      --work-dir=<workDir>                 Path where the pipeline scratch data is stored.
  -p, --profile=<profile>[,<profile>...]   Comma-separated list of one or more configuration profile names you want to use for this pipeline execution.
  -r, --revision=<revision>                A valid repository commit Id, tag or branch name.
      --wait=<wait>                        Wait until given status or fail. Valid options: SUBMITTED, RUNNING, SUCCEEDED, FAILED, CANCELLED, UNKNOWN.
  -h, --help                               Show this help message and exit.
  -V, --version                            Print version information and exit.

Advanced options:
      --config=<config>                    Additional Nextflow config file.
      --pre-run=<preRunScript>             Bash script that is executed in the same environment where Nextflow runs just before the pipeline is launched.
      --post-run=<postRunScript>           Bash script that is executed in the same environment where Nextflow runs immediately after the pipeline completion.
      --pull-latest                        Enable Nextflow to pull the latest repository version before running the pipeline.
      --stub-run                           Execute the workflow replacing process scripts with command stubs.
      --main-script=<mainScript>           Pipeline main script file if different from `main.nf`.
      --entry-name=<entryName>             Main workflow name to be executed when using DLS2 syntax.
      --schema-name=<schemaName>           Schema name.
      --user-secrets=<userSecrets>[,<userSecrets>...]
                                           Pipeline Secrets required by the pipeline execution that belong to the launching user personal context. User's
                                             secrets will take precedence over workspace secrets with the same name.
      --workspace-secrets=<workspaceSecrets>[,<workspaceSecrets>...]
                                           Pipeline Secrets required by the pipeline execution. Those secrets must be defined in the launching workspace.
```

Versions:
```
     Tower API version       | 1.19.0
     Tower version           | 23.2.0-cycle12
     CLI version             | 0.8.0 (3582751)
     CLI minimum API version | 1.15
```

### PR DESCRIPTION
**Pull request by JaimeSeqLabs**
## Description

Introduce a new flag to add labels when launching pipelines.

Closes #312 

## Guidelines for testing

+ Launch a pipeline adding labels:
`tw launch test_hello -l CliLabelOne,CliLabelTwo,UserLabel -w Org/Wsp`

Any label that does't exist previous to the launch will be created

+ List pipeline runs showing labels:
`tw runs list -l -w Org/Wsp`

image

### CODE REVIEW

File: src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java [hunk 1]
```diff
@@ -209,6 +229,49 @@ private WorkflowStatus checkWorkflowStatus(String workflowId, Long workspaceId)
...
+            .stream()
+            .collect(Collectors.toMap(LabelDbDto::getName, LabelDbDto::getId));
+
+        // get label names not registered in workspace (names are unique per wspID)
+        List<String> newLabels = labels
+            .stream()
+            .filter(labelName -> !nameToID.containsKey(labelName))
+            .collect(Collectors.toList());
+
+        // create the new ones via POST /labels
```
Comment (tcrespog) [inline src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java:256]: Workspace participants with launch role cannot create labels (only use existing ones). For that reason, I don't know to what extent creating labels on the fly should be supported :thinking:.

Comment (gwright99) [reply src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java:256]: Seems unproductive to me that a Launch user can't create a metadata tag for the pipeline they are invoking.

File: src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java [hunk 2]
```diff
@@ -209,6 +240,88 @@ private WorkflowStatus checkWorkflowStatus(String workflowId, Long workspaceId)
         }
     }
 
+    private List<Long> obtainLabelIDs(@Nullable Long workspaceId) throws ApiException {
+
+        if (labels == null || labels.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        if (!labelPermission(workspaceId)) {
```
Comment (tcrespog) [inline src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java:249]: Note `launch` users cannot modify the given pipeline labels, but they must send those. According to this code, `launch` user won't be able to launch pipelines with labels. To summarize:
- Launch users must stick to the labels of the pipeline.
- Launch users cannot create new labels.

File: src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java [hunk 3]
```diff
@@ -209,6 +240,88 @@ private WorkflowStatus checkWorkflowStatus(String workflowId, Long workspaceId)
...
+
+        // personal workspace
+        if (wspId == null) return true;
+
+        var client = HttpClient.newBuilder()
+            .connectTimeout(Duration.ofSeconds(10))
+            .build();
+
+        var uri = UriBuilder
+            .fromUri(URI.create(apiUrl() + "/permissions"))
```
Comment (jordeu) [inline src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java:300]: Shouldn't this be part of the SDK? The rest of the CLI is always using the Tower SDK to call Tower API.

Comment (jordeu) [reply src/main/java/io/seqera/tower/cli/commands/LaunchCmd.java:300]: Is something blocking to add the `/permissions` endpoint to the SDK? But nevermind, as you wish, but this is the typical thing that theoretically it's temporal but it ends up here forever. So at least add a big TODO comment here explaining why this was done like this and when it should be removed.

**Review by tcrespog — APPROVED** [review]

**Pull request merged**